library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopEdgeTest.java

```diff
@@ -112,4 +156,60 @@ public void shouldCreateReadableToString() {
         // Then
         assertEquals("e[source-BasicEdge->dest]", toString);
     }
+
+    @Test
+    public void shouldReturnOutVertex() {
+        // Given
+        final String source = "source";
+        final String dest = "dest";
+        final GafferPopGraph graph = mock(GafferPopGraph.class);
+        final GafferPopVertex outVertex = new GafferPopVertex(GafferPopGraph.ID_LABEL, source, graph);
+        final GafferPopVertex inVertex = new GafferPopVertex(GafferPopGraph.ID_LABEL, dest, graph);
+        final List<Vertex> result = new ArrayList<>();
+
+        // When
+        final GafferPopEdge edge = new GafferPopEdge(TestGroups.EDGE, outVertex, inVertex, graph);
+
+        // Then
+        edge.vertices(Direction.OUT).forEachRemaining(result::add);
+        assertThat(result).contains(outVertex);
```

Comment:
For iterables we usually use the Assertj assertions directly with `containsExactly` and similar assertions.
```suggestion
        final GafferPopVertex inVertex = new GafferPopVertex(GafferPopGraph.ID_LABEL, dest, graph);

        // When
        final GafferPopEdge edge = new GafferPopEdge(TestGroups.EDGE, outVertex, inVertex, graph);

        // Then
        assertThat(edge.vertices(Direction.OUT)).toIterable().containsExactly(outVertex);
```